library/src/main/java/com/bumptech/glide/load/resource/bitmap/TransformationUtils.java

```diff
@@ -139,6 +139,55 @@ public static Bitmap fitCenter(@NonNull BitmapPool pool, @NonNull Bitmap inBitma
   }
 
   /**
+   * Operation to center the given Bitmap into the given target. If the Bitmap is larger than the
+   * target, {@link #fitCenter(BitmapPool, Bitmap, int, int)} is called instead.
+   *
+   * @param pool   The BitmapPool obtain a bitmap from.
+   * @param inBitmap  The Bitmap to center.
+   * @param width  The width in pixels of the target.
+   * @param height The height in pixels of the target.
+   * @return A new Bitmap centered within the given dimensions, or toFit if the Bitmap's width or
+   * height is larger than the given dimensions
+   */
+  public static Bitmap centerInside(@NonNull BitmapPool pool, @NonNull Bitmap inBitmap, int width,
+                                 int height) {
+    if (inBitmap.getWidth() == width && inBitmap.getHeight() == height) {
+      if (Log.isLoggable(TAG, Log.VERBOSE)) {
+        Log.v(TAG, "requested target size matches input, returning input");
+      }
+      return inBitmap;
+    }
+
+    if (inBitmap.getWidth() > width || inBitmap.getHeight() > height) {
+      if (Log.isLoggable(TAG, Log.VERBOSE)) {
+        Log.v(TAG, "requested target size too big for input, fit centering instead");
+      }
+      return fitCenter(pool, inBitmap, width, height);
+    }
+    final int horizontalPadding = (width - inBitmap.getWidth()) / 2;
+    final int verticalPadding = (height - inBitmap.getHeight()) / 2;
+
+    Bitmap.Config config = getSafeConfig(inBitmap);
```

Comment:
I think this needs to upgrade to 8888 if there's a padding, otherwise we'll end up with a color that someone won't like (BLACK/WHITE are both wrong choices), it should be TRANSPARENT.